src/components/Switch/Switch.tsx

```diff
@@ -1,21 +1,27 @@
 import { Switch as BaseSwitch } from '@kobalte/core';
-import { Component, ComponentProps, ParentComponent } from 'solid-js';
+import { Component, ComponentProps, ParentComponent, splitProps } from 'solid-js';
 
 import * as styles from '@cubeartisan/cubeartisan/components/Switch/Switch.css';
 
 const SwitchInput = BaseSwitch.Input;
 
-export const SwitchLabel: ParentComponent<ComponentProps<typeof BaseSwitch.Label>> = (props) => (
-  <BaseSwitch.Label class={styles.switchLabel} {...props} />
-);
+const SwitchLabel: ParentComponent<ComponentProps<typeof BaseSwitch.Label>> = (props) => {
+  const [local, others] = splitProps(props, ['class']);
 
-const SwitchControl: ParentComponent<ComponentProps<typeof BaseSwitch.Control>> = (props) => (
-  <BaseSwitch.Control class={styles.switchControl} {...props} />
-);
+  return <BaseSwitch.Label class={`${styles.switchLabel} ${local.class}`} {...others} />;
+};
+
+const SwitchControl: ParentComponent<ComponentProps<typeof BaseSwitch.Control>> = (props) => {
+  const [local, others] = splitProps(props, ['class']);
+
+  return <BaseSwitch.Control class={`${styles.switchControl} ${local.class}`} {...others} />;
```

Comment:
If there isn't a class in the passed in props will this fill in undefined or just a trailing space?

Reply:
I don't know actually